### PR TITLE
fix(web): removed double encoding of CSS issue

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

 No Github issues to link

## Intent

Summary of issues:
- `sasjs deploy` is throwing errors (parsing log error for `SAS9`)
- `sasjs web` should really be in `sasjs compile` not `sasjs build`
- CSS is being base64 encoded twice
- add mime type to use for OGG

## Implementation

Moved code for processing for `webStream` from `build.ts` to `compile.ts`
Updated `deploy.ts` for `executeScriptSAS9` result log.
Updated `web/index.ts`
- to avoid encoding of content twice 
Updated `sasjsout.ts`
- to support OGG file as well 

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [ ] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server:
image
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

**Comment by allanbowe** [comment]
windows tests passing
image

**Review by allanbowe — CHANGES_REQUESTED** [review]
image

The use of windows slashes in the generated SAS program is not valid for SAS logical folder paths

image


In the generated SAS program, we must always use forward slashes (`/`) to as a folder seperator
****

**Comment by saadjutt01** [comment]
> image
> 
> The use of windows slashes in the generated SAS program is not valid for SAS logical folder paths
> 
> image
> 
> In the generated SAS program, we must always use forward slashes (`/`) to as a folder seperator

Fixed!
image

**Comment by allanbowe** [comment]
Feature is working now in SAS 9:
image

File: src/commands/web/index.ts [hunk 1]
```diff
@@ -270,15 +271,21 @@ async function createAssetsServicesNested(
 }
 
 async function generateAssetService(
-  content: string,
+  sourcePath: string,
   filePath: string,
   destinationPath: string,
   serverType: ServerType
 ) {
+  const nonBinaryFileTypes = ['html', 'CSS', 'JS']
```
Comment (krishna-acondy) [inline src/commands/web/index.ts:279]: Shouldn't `html` be uppercased as well?

Comment (allanbowe) [reply src/commands/web/index.ts]: uppercased

File: src/commands/build/build.ts [hunk 2]
```diff
@@ -250,7 +235,7 @@ async function getContentFor(
       ? ''
       : testPath !== undefined
       ? testPath
-      : folderPath.replace(rootDirectory, '').substr(1)
+      : folderPath.replace(rootDirectory, '').split('\\').splice(1).join('/')
```
Comment (YuryShkoda) [inline src/commands/build/build.ts:238]: use `path.sep` to make it more robust

**Review by allanbowe — CHANGES_REQUESTED** [review]
Looks like the windows path update broke the non-windows build for Data Controller:
image

image

File: src/commands/build/build.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/build/build.ts]: done

**Review by allanbowe — CHANGES_REQUESTED** [review]
When building Data Controller, we are getting the following error in console:

image

**Comment by saadjutt01** [comment]
> When building Data Controller, we are getting the following error in console:
> 
> image

Issue is resolved.

**Comment by allanbowe** [comment]
Data Controller is building fine now

**Review by krishna-acondy — APPROVED** [review]

**Review by allanbowe — APPROVED** [review]

**Pull request merged**